src/language_concepts/data_types.md

```diff
@@ -1,206 +1,192 @@
+> **Note:** When returning a boolean value, it will show up as a value of 1 for `true` and 0 for `false` in _Verifier.toml_.
 
-> **Note:** This behaviour and type will change in future releases, to catch a linearity bug in user code.
+The main usage of the boolean type is in conditionals (e.g. `if` expressions, `constrain`). More about conditionals are covered in the [Control Flow](./control_flow.md) and [Constrain Statement](./constrain.md) sections.
 
 ## Compound Types
 
-Noir also supports arrays, structs, and tuples whose elements may be public or private.
+A compound type groups together multiple values into one type. Elements within a compound type can be private or public.
 
-#### Arrays
+### The Array Type
 
-An array is a data structure which allows you to group together data types.
-All values in an array must be of the same type (i.e., they're homogeneous).
+An array is one way of grouping together values into one compound type:
 
 ```rust,noplaypen
 fn main(x : Field, y : Field) {
     let my_arr = [x, y];
 }
 ```
 
-Note that currently Noir only supports arrays with integer or `Field` elements.
+All elements in an array must be of the same type (i.e. homogeneous). That is, `Field` values cannot be grouped together with integer values in one array for example.
 
-> **Example:** An array of `Field` types cannot be grouped together with an array of integer types.
+Noir currently supports arrays with `Field` or integer elements only.
 
-#### Structs
+### The Tuple Type
 
-Structs can be used to group together multiple different types:
+A tuple collects multiple values like an array, but with the added ability to collect values of different types:
 
 ```rust,noplaypen
-struct Operation {
-    lhs: Field,
-    opcode: u8,
-    rhs: Field,
+fn main() {
+    let tup: (u8, u64, Field) = (255, 500, 1000);
 }
+```
+
+One way to access tuple elements is via destructuring using pattern matching:
 
+```rust,noplaypen
 fn main() {
-    let opcode = 3 as u8;
+    let tup = (1, 2);
 
-    // operation: Operation
-    let operation = Operation {
-        lhs: 0,
-        rhs: 1,
-        opcode,
-    }
+    let (one, two) = tup;
 
-    let zero = operation.lhs;
+    let three = one + two;
 }
 ```
 
-Structs can also be destructured in a pattern, binding each field to a new variable:
+Another way to access tuple elements is via direct member access, using a period (`.`) followed by the index of the element we want to access. Index `0` corresponds to the first tuple element, `1` to the second and so on:
 
 ```rust,noplaypen
 fn main() {
-    let Operation { lhs, rhs: the_rhs, opcode: _ } = get_operation();
-    // lhs and the_rhs are now in scope
+    let tup = (5, 6, 7, 8);
+
+    let five = tup.0;
+    let eight = tup.3;
 }
 ```
 
-#### Tuples
+### Structs
+
+A struct is used for grouping multiple related values of different types like a tuple, with the added ability to name the pieces of data (also referred to as _fields_).
```

Comment:
This reads a bit awkwardly with the ", with" and no conjunction. Perhaps we can rephrase it:
"A struct also allows for grouping multiple values of different types. Unlike tuples, we can also name each field. Note that the usage of "field" here is referring to each element of the struct and is unrelated to the `Field` type."

So we can merge the note below into the paragraph itself. I've also removed the mention of Rust since the phrase "a struct's fields" does not originate with rust nor was it popularized by it. If we want to keep the reasoning though we could mention that this follows convention established by other programming languages.